packages/salesforcedx-vscode-apex/src/views/testOutline.ts

```diff
@@ -0,0 +1,666 @@
+const taskViewService = sfdxCoreExports.taskViewService;
+
+const LIGHT_BLUE_BUTTON = ospath.join(
+  __filename,
+  '..',
+  '..',
+  '..',
+  '..',
+  'resources',
+  'light',
+  'testNotRun.svg'
+);
+const LIGHT_RED_BUTTON = ospath.join(
+  __filename,
+  '..',
+  '..',
+  '..',
+  '..',
+  'resources',
+  'light',
+  'testFail.svg'
+);
+const LIGHT_GREEN_BUTTON = ospath.join(
+  __filename,
+  '..',
+  '..',
+  '..',
+  '..',
+  'resources',
+  'light',
+  'testPass.svg'
+);
+
+const DARK_BLUE_BUTTON = ospath.join(
+  __filename,
+  '..',
+  '..',
+  '..',
+  '..',
+  'resources',
+  'dark',
+  'testNotRun.svg'
+);
+const DARK_RED_BUTTON = ospath.join(
+  __filename,
+  '..',
+  '..',
+  '..',
+  '..',
+  'resources',
+  'dark',
+  'testFail.svg'
+);
+const DARK_GREEN_BUTTON = ospath.join(
+  __filename,
+  '..',
+  '..',
+  '..',
+  '..',
+  'resources',
+  'dark',
+  'testPass.svg'
+);
+
+export type FullTestResult = {
+  summary: TestSummary;
+  tests: TestResult[];
+};
+
+export type TestSummary = {
+  outcome: string;
+  testsRan: number;
+  passing: number;
+  failing: number;
+  skipped: number;
+  passRate: string;
+  failRate: string;
+  testStartTime: string;
+  testExecutionTime: string;
+  testTotalTime: string;
+  commandTime: string;
+  hostname: string;
+  orgId: string;
+  username: string;
+  testRunId: string;
+  userId: string;
+};
+
+export type TestResult = {
+  ApexClass: ApexClass;
+  MethodName: string;
+  Outcome: string;
+  RunTime: number;
+  Message: string;
+  StackTrace: string;
+  FullName: string;
+};
+
+export type ApexClass = {
+  attributes: { type: string };
+  Id: string;
+  Name: string;
+  NamespacPrefix: string;
+};
+
+export type ApexGroupInfo = {
+  outcome: string;
+  summary: TestSummary;
+};
+
+export class ApexTestOutlineProvider implements vscode.TreeDataProvider<Test> {
+  private _onDidChangeTreeData: vscode.EventEmitter<
+    Test | undefined
+  > = new vscode.EventEmitter<Test | undefined>();
+  public readonly onDidChangeTreeData: vscode.Event<Test | undefined> = this
+    ._onDidChangeTreeData.event;
+
+  private eventsEmitter = new events.EventEmitter();
+
+  private apexTestMap: Map<string, Test> = new Map<string, Test>();
+  private head: Test | null;
+  private static testStrings: Set<string> = new Set<string>();
+
+  constructor(
+    private path: string,
+    private apexClasses: vscode.Uri[],
+    private apexTestInfo: ApexTestRequestInfo[] | null
+  ) {
+    this.head = null;
+    this.getAllApexTests(this.path);
+    // Now, activate events that we need
+    this.eventsEmitter.on('Delete Folder', this.deleteFolderRecursive); // Activate event to delete folder
+    this.eventsEmitter.on('Show Highlight', this.updateSelection);
+  }
+
+  public getHead(): Test {
+    if (this.head === null) {
+      return this.getAllApexTests(this.path);
+    }
+    return this.head;
+  }
+
+  public getChildren(element: Test): Thenable<Test[]> {
```

Comment:
Must we use `Thenable`? Can we use a combination of async/await? Shouldn't mix different styles of programming paradigms if possible.